### PR TITLE
net: mgmt: Fix the maximum event size calculation

### PR DESCRIPTION
**Pull request by krish2718**
When DHCPv4 is enabled it is the highest size, so, use that as a first choice. Also, Update the stale comment.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]
Looks good.

IMHO, the whole info size check is quite cumbersome and a bit error prone. So I am pondering here that should we declare the size to some arbitrary value (perhaps 64?), and then just trying to make sure in compile time checks that all the needed structs fit into this space. This would waste some bytes, but perhaps that is an acceptable tradeoff.

**Comment by krish2718** [comment]
> Looks good.
> 
> IMHO, the whole info size check is quite cumbersome and a bit error prone. So I am pondering here that should we declare the size to some arbitrary value (perhaps 64?), and then just trying to make sure in compile time checks that all the needed structs fit into this space. This would waste some bytes, but perhaps that is an acceptable tradeoff.

Definitely agree, current approach is not maintainable, 64 seems reasonable. Is dynamic memory allocation an option (HEAP/static pools)?

**Comment by krish2718** [comment]
Just to be clear, I am proposing we should merge this first and then work on improvements in the background.

**Review by jukkar — DISMISSED** [review]

**Review by mniestroj — COMMENTED** [review]
How about defining a union with all those structs inside, then doing sizeof on that union? This would be future proof to introducing new members in structs.

**Comment by krish2718** [comment]
> How about defining a union with all those structs inside, then doing sizeof on that union? This would be future proof to introducing new members in structs.

Sure, do you want me to do it in this PR itself or submit another one?

File: subsys/net/ip/net_mgmt.c [hunk 1]
```diff
@@ -67,8 +67,8 @@ static inline void mgmt_push_event(uint32_t mgmt_event, struct net_if *iface,
 			memcpy(events[i_idx].info, info, length);
 			events[i_idx].info_length = length;
 		} else {
-			NET_ERR("Event info length %zu > max size %zu",
-				length, NET_EVENT_INFO_MAX_SIZE);
+			NET_ERR("Event %d info length %zu > max size %zu",
```
Comment (jukkar) [inline subsys/net/ip/net_mgmt.c:70]: mgmt_event var is unsigned, so `%u` would be more apppropriate here.

**Review by mniestroj — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**